File: C++/Problems/Problems/problem_3120.cpp
#include "problem_3120.h"

#include <unordered_map>

bool problem_3120::test() {
    auto word = "aaAbcBC";

    auto expected = 3;

    auto reslut = numberOfSpecialChars(word);

    return expected == reslut;
}

struct Case
{
    bool Lower;
    bool Upper;
};

int problem_3120::numberOfSpecialChars(std::string word) {
    std::unordered_map<int, Case> charDic;
    for (int i = 0; i < 26; ++i)
        charDic[i] = Case();

    for (auto c : word)
    {
        if (std::isupper(c))
            charDic[c - 'A'].Upper = true;
        else
            charDic[c - 'a'].Lower = true;
    }

    auto result = 0;
    for (auto &c : charDic)
        if (c.second.Upper && c.second.Lower)
            result++;

    return result;
}

File: Python/Problems/Problems/problem3120.py
from .problem import Problem

class Problem3120(Problem):
    def test(self):
        word = "aaAbcBC"

        expected = 3

        reslut = self.numberOfSpecialChars(word)

        return expected == reslut

    def numberOfSpecialChars(self, word):
        charDic = {}
        for i in range(26):
            charDic[i] = [False, False]

        for c in word:
            if c.isupper():
                charDic[ord(c) - ord('A')][0] = True
            else:
                charDic[ord(c) - ord('a')][1] = True

        result = 0
        for c in charDic.values():
            if c[0] and c[1]:
                result += 1

        return result